historical-adherence-query/java/Main.java

```diff
@@ -0,0 +1,203 @@
+import com.google.common.base.Stopwatch;
+import com.mypurecloud.sdk.v2.ApiClient;
+import com.mypurecloud.sdk.v2.ApiException;
+import com.mypurecloud.sdk.v2.Configuration;
+import com.mypurecloud.sdk.v2.PureCloudRegionHosts;
+import com.mypurecloud.sdk.v2.api.UsersApi;
+import com.mypurecloud.sdk.v2.api.WorkforceManagementApi;
+import com.mypurecloud.sdk.v2.extensions.notifications.NotificationHandler;
+import com.mypurecloud.sdk.v2.extensions.notifications.NotificationListener;
+import com.mypurecloud.sdk.v2.model.UserMe;
+import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice;
+import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceQuery;
+import com.mypurecloud.sdk.v2.model.WfmHistoricalAdherenceResponse;
+import com.neovisionaries.ws.client.WebSocketException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.function.Supplier;
+
+public class Main {
+
+    public static void main(String[] args) throws IOException, ApiException, InterruptedException, WebSocketException,
+        TimeoutException {
+
+        // For more information on how to authenticate your api client, here is a useful resource.
+        // https://developer.mypurecloud.com/api/rest/client-libraries/java/index.html#authenticating
+        String clientId = System.getenv("PURECLOUD_CLIENT_ID");
+        String clientSecret = System.getenv("PURECLOUD_CLIENT_SECRET");
+
+        // Set Region
+        PureCloudRegionHosts region = PureCloudRegionHosts.us_east_1;
+        ApiClient apiClient = ApiClient.Builder.standard()
+            .withBasePath(region)
+            .build();
+
+        ApiResponse<AuthResponse> authResponse = apiClient.authorizeClientCredentials(clientId, clientSecret);
+
+        // Set the ApiClient instance as your default api client
+        Configuration.setDefaultApiClient(apiClient);
+
+        // Create an instance of UsersApi to retrieve your own userId
+        UsersApi usersApi = new UsersApi();
+        UserMe me = usersApi.getUsersMe(Collections.singletonList("presence"));
```

Comment:
As a best practice, don't expand anything if you aren't using it; using expands causes additional backend service requests to populate the requested data.